### ISSUE TITLE
Outdated package key for NVIDIA CUDA

### ISSUE THREAD

**Issue by mabelzhang** [Field-Robotics-Lab/dockwater#24]
Cause:
"NVIDIA is updating and rotating the signing keys used by apt, dnf/yum, and zypper package managers beginning April 27, 2022. Failure to update your repository signing keys will result in package management errors when attempting to access or install packages from CUDA repositories."
https://forums.developer.nvidia.com/t/notice-cuda-linux-repository-key-rotation/212772

Symptom:
Running (`run.bash`) the Noetic image gives this error:
```
building > W: GPG error: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A4B469963BF863CC
E: The repository 'https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease' is not signed.
```

Solution:
See link above.

I haven't had time to fix it. Just putting up this for awareness.

**Comment by woensug-choi** [comment]
I bumped into this. I had to fix rocker to solve.

**Comment by mabelzhang** [comment]
Oh does it have to do with the --nvidia flag to Rocker? Worth a ticket in Rocker?

**Comment by woensug-choi** [comment]
I am bit foncused about this issue too. Took quite some time to make the multibeam sonar to run (this case, using cuda-dev branch of the rocker which contained old key installation line)

What I did was first was to add those new key install commands in dockwater(failed). Second, replace key installation line in cuda-dwv branch of rocker (failed, kept calling old lines even after reinstallation of rocker). Removed all docker images (failed, no change). Uses pip virtual env (venv, howto is at rocker readme) was a final success.

Yes, I suspect the solution has to be made on the rocker cuda flag
